### ISSUE TITLE
508-defect-2 [KEYBOARD]: BAM2 Debt Letters - The debt download table sorting function must be keyboard accessible

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#17894]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Josh_

## User Story or Problem Statement

As a screen reader user, I am confused when the download debt letters table headers announce themselves as "clickable", but when I press Enter or Space, nothing happens. As a keyboard user, I would like to sort the table by column, but the table headers will not receive keyboard focus.

## Details

The download debt letters table columns 1 and 2 have icons that allow users to click and sort on those columns. This feature must be accessible to keyboard and screen reader users. We should add a button inside the TH, and listen for click events on that instead of the TH. Code snippet below.

## Related Issues
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/16789

## Acceptance Criteria

- [ ] Tabindex -1 is removed from all TH columns
- [ ] Clickable column headers have a button inside the TH
- [ ] Visual styles are maintained
- [ ] Ideally the chevron icon is added to the button as a background image instead of an `<i>` element. If the `<i>` is maintained, it must have an `aria-hidden="true"` attribute applied.
- [ ] No axe-core violations
- [ ] Any sorting function has been discussed and collaborated with the accessibility specialist

## Fix (If known)
```diff
<th
  class="vads-u-border--0"
  scope="row"
- tabindex="-1"
>
+   <button type="button" class="CLASSES">Type</button>
-   <i class="fas fa-sort vads-u-margin-left--0p5"></i>
</th>
```

## Environment

* https://staging.va.gov/manage-va-debt/your-debt/debt-letters
* Test user 38

## WCAG or Vendor Guidance (optional)
* [Keyboard: Understanding SC 2.1.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-keyboard-operable.html)

## Screenshots or Trace Logs

![Screen Shot 2020-12-28 at 5 24 52 PM](https://user-images.githubusercontent.com/934879/103248799-e2123000-4931-11eb-803d-6349a66a212e.png)

**Comment by joshkimux** [comment]
Hi @denisecoveyduc , has there been any plans to try and tackle this issue? If y'all need support, you could loop in CAIA

**Comment by denisecoveyduc** [comment]
The debt portal is no longer offering Debt Letters for download due to a privacy incident that caused VBA. As a result this accessibility is no longer an issue.
